core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainerFactory.scala

```diff
@@ -17,30 +17,35 @@
 
 package whisk.core.containerpool.docker
 
+import java.nio.file.Paths
+import java.util.concurrent.TimeoutException
+
 import akka.actor.ActorSystem
-import scala.concurrent.Await
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future
 import whisk.common.Logging
 import whisk.common.TransactionId
-import whisk.core.WhiskConfig
 import whisk.core.containerpool.Container
 import whisk.core.containerpool.ContainerFactory
 import whisk.core.containerpool.ContainerFactoryProvider
 import whisk.core.entity.ByteSize
 import whisk.core.entity.ExecManifest
 import whisk.core.entity.InstanceId
+import whisk.core.ConfigKeys
+import whisk.core.WhiskConfig
+
 import scala.concurrent.duration._
-import java.util.concurrent.TimeoutException
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
 
 class DockerContainerFactory(config: WhiskConfig, instance: InstanceId, parameters: Map[String, Set[String]])(
   implicit actorSystem: ActorSystem,
   ec: ExecutionContext,
   logging: Logging)
     extends ContainerFactory {
+  private val dockerConfig = pureconfig.loadConfigOrThrow[DockerContainerFactoryConfig](ConfigKeys.invokerDocker)
```

Comment:
Would like some advice on if this is the best pattern for the pureconfig setup. I think the DockerContainter factory is probably the right spot for reading some config but not sure how much I should be pulling into the case class and if the "key" is appropriate in WhiskConfig.scala.

Reply:
It seems at least that the above example works just fine -- meaning the case class just ignores "extra" parameters, quick test proves this out:

```scala
import com.typesafe.config.ConfigFactory
import pureconfig._
import scala.concurrent.duration._
import com.typesafe.config.ConfigFactory
case class Holiday(where: String)
loadConfig[Holiday](ConfigFactory.parseString("{ where: cabo }")) // Works
loadConfig[Holiday](ConfigFactory.parseString("{ where: cabo, why: hagar }")) //Works just "cabo" available
loadConfig[Holiday](ConfigFactory.parseString("{ why: hagar }")) // "fails" because 'where' not defined.
```